src/main.rs

```diff
@@ -154,204 +159,254 @@ async fn main() -> Result<(), Box<dyn Error>> {
     // Use the new non-global metrics registry when we upgrade to newer version of malachite
     let _ = Metrics::register(registry);
     let (messages_request_tx, messages_request_rx) = mpsc::channel(100);
-    let (shard_decision_tx, shard_decision_rx) = broadcast::channel(100);
 
-    let global_db = RocksDB::open_global_db(&app_config.rocksdb_dir);
-    let local_state_store = LocalStateStore::new(global_db);
+    if app_config.read_node {
```

Comment:
main is getting rather large, and now with two distinct branches it's harder to reason about. Worth splitting it into two, but can happen in a followup

Reply:
Yeah, agreed that this is messy. Let's address in a follow up. There are significant changes here for supporting rpcs + mempool so would be good to evaluate what to do after those changes.